src/client.rs

```diff
@@ -113,14 +118,22 @@ impl Client {
     }
 
     // Request must be signed
-    fn sign_request(&self, endpoint: &str, request: &str) -> String {
-        let mut signed_key = Hmac::<Sha256>::new_varkey(self.secret_key.as_bytes()).unwrap();
-        signed_key.update(request.as_bytes());
-        let signature = hex_encode(signed_key.finalize().into_bytes());
-        let request_body: String = format!("{}&signature={}", request, signature);
-        let url: String = format!("{}{}?{}", self.host, endpoint, request_body);
-
-        url
+    fn sign_request(&self, endpoint: API, request: Option<String>) -> String {
+        match request {
+            Some(request) => {
+                let mut signed_key = Hmac::<Sha256>::new_varkey(self.secret_key.as_bytes()).unwrap();
+                signed_key.update(request.as_bytes());
```

Comment:
I guess you can add this line to the previous, and then you can drop the `mut` on the `signed_key`. Not sure what is better Rust practice :)

Reply:
Ok, we can discuss this later. Could you please review the #83 first?